core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -458,7 +458,9 @@ object KafkaConfig {
   val AzpubsubClientCertificateAclProp = "azpubsub.client.ceritificate.acl"
```

Comment:
no specific config addition here... we should have config impl in azpubsub package

Reply:
No. In order to use server.properties to configure appki client cert acls, need to have this code in place. And this is not something new - it was merge a couple of months back.

Reply:
as i explained offline, we should add the logic to have the config specific to azpubsub package and should not integrate that code into core Kafka code base. this will leads to more issues in future (like the same explanation that you provided above saying it is added previously and wanted to add more). everyone will follow the same trend and it will get messed up.

we should own, clean it up and make the azpubsub & azpubsub-api packages as really 100% plugin model. take a look at the the log4j on how they have the config handled outside of Kafka and that is 100% plugin model.

Also understand it wont come for free. we should invest and get it clean which will add value to the product and the team. please work on it and let me know if you need any guidance.

Note: please dont close the review abruptly without satisfying the reviewers feedback as each reviewer will have some value. thnx!

Reply:
@senthilm-ms , thanks for the comments.  Firstly I wanted to clarify something before further discussion. 
there're two kinds of configuration settings: 
- settings  that are used by azpubsub modules, this is something like log4j.properties used by log4j modules; 
- settings that are used to load azpubsub modules.  

Your log4j example makes sense to me now. I will figure out a way to do this module specific configuration inside of the azpubsub module instead of kafka server.properties. 

But for second kind of settings that are used to load azpubsub modules, based on my investigation, we will have to make minor code change in KafkaConfig class to make it happen. 
I thought you wanted to read all the configs from server.properties without a single line of code change, that should be not possible. But your thought makes sense to me know I will figure this out. 

Thanks!

Reply:
@senthilm-ms, I addressed this review comment: 

- the KafkaConfig.scala is reverted - all dsts relevant settings go to a separate file "azpubsub.properties". 
- The SASL OAuth classese are also reverted, separate classes are created in "azpubsub" module. The original Kafka code is cleaned up. 
- Unit test added or updated.

Please continue to review.